app/models/schedule.rb

```diff
@@ -2,13 +2,22 @@ class Schedule < ApplicationRecord
   belongs_to :user
   has_many :slots, inverse_of: :schedule, dependent: :destroy
   accepts_nested_attributes_for :slots, allow_destroy: true
-  scope :by_from, -> { order(:from) }
+  scope :by_start_at, -> { order(:start_at) }
 
-  validates :from, presence: true
-  validate :from_must_be_more_than_six_weeks_in_the_future, unless: :first_schedule?
+  validates :start_at, presence: true
+  validate :start_at_must_be_more_than_six_weeks_in_the_future, unless: :first_schedule?
+
+  def end_at
+    self[:end_at] - 1.day if self[:end_at]
+  end
+
+  def self.with_end_at
+    select('*, LEAD(start_at, 1) OVER (ORDER BY start_at) AS end_at')
+      .order(:start_at)
```

Comment:
I know it's a little late to the party but changing it to the following should remove any future issues with guider scope:
```
LEAD(start_at) OVER (PARTITION BY guider_id ORDER BY start_at) AS end_at
```